### DISCUSSION TITLE
Is the structure of subgenerators a bit fragmented now?

### DISCUSSION THREAD

**Discussion by wangxinxx** [jhipster/generator-jhipster#25972] [Q&A]
##### **Overview of the feature request**

At the next level of the java sub-generator, there are also generators such as jib/bootstrap/build-tool/code-quality. Among the templates of these generators, there are only 1-2 template files. There are currently 15 generators involving the Java language, and I feel that it brings challenges to the production of blueprints.

##### **Related issues or PR**

- [ ] Checking this box is mandatory (this is just to show you read everything)

**Comment by mshima** [comment]
The idea is to have them reusable out-of-the-box.
Micronaut blueprint:
https://github.com/jhipster/generator-jhipster-micronaut/blob/3b98726fe7ff444d8bb4bbe8cb02152421f91c5a/generators/micronaut/generator.js#L24
https://github.com/jhipster/generator-jhipster-micronaut/blob/3b98726fe7ff444d8bb4bbe8cb02152421f91c5a/generators/micronaut/generator.js#L59-L65

I don’t know your use case, but for official blueprints, it improved reusability.

**Comment by wangxinxx** [comment]
OK. Do see an improvement in reusability. The challenge I encountered is: in sbs mode, I need to constantly adjust the directories or files where I add templates, and adjust the codes of multiple sub-generators at the same time. In addition, there is also an uncertain factor. How many such adjustments may there be in the future? Will there be other more adjustments to the position of the sub-generator?

**Comment by mshima** [comment]
If you provide some insight of what customizations you make, we can try to avoid such changes.
Almost every change can  break blueprints.

**Comment by TatyOko28** [comment]
## Overview of the feature request

I would like to propose a simplification/refactoring of the Java sub-generator structure.

At the next level of the Java sub-generator, there are many small generators such as:

```text
jib
bootstrap
build-tool
code-quality
...
